src/sv-harvester-job.ts

```diff
@@ -0,0 +1,170 @@
+
+    try {
+      // Get the signer's (keeper) current nonce.
+      const currentNonce = await provider.getTransactionCount(txSigner.address);
+
+      // Fetch the priorityFeeInGwei and maxFeePerGas parameters from the getMainnetGasType2Parameters function
+      // NOTE: this just returns our priorityFee in GWEI, it doesn't calculate it, so if we pass a priority fee of 10 wei
+      //       this will return a priority fee of 10 GWEI. We need to pass it so that it properly calculated the maxFeePerGas
+      const { priorityFeeInGwei, maxFeePerGas } = getMainnetGasType2Parameters({
+        block,
+        blocksAhead,
+        priorityFeeInWei: PRIORITY_FEE,
+      });
+
+      // We declare what options we would like our transaction to have
+      const options = {
+        gasLimit: 10_000_000,
+        nonce: currentNonce,
+        maxFeePerGas,
+        maxPriorityFeePerGas: priorityFeeInGwei,
+        type: 2,
+      };
+
+      // We populate the transactions we will use in our bundles
+      let txs: TransactionRequest[];
+
+      // Check if account settlement is required
+      const isAccountSettlementRequired = await job.isAccountSettlementRequired(vault.address);
+
+      // If it is, then we will send a transaction to the job to settle the account
+      if (isAccountSettlementRequired) {
+        console.info('Account settlement is required!');
+        txs = await populateTransactions({
+          chainId: CHAIN_ID,
+          contract: job as Contract,
+          functionArgs: [[vault.address]],
+          functionName: 'settleAccount',
+          options,
+        });
+      } else {
+        // If it's not, then we will send a transaction to the job to harvest
+        txs = await populateTransactions({
+          chainId: CHAIN_ID,
+          contract: job as Contract,
+          functionArgs: [[vault.address]],
+          functionName: 'harvest',
+          options,
+        });
+      }
+
+      // We calculate the first block that the first bundle in our batch will target.
+      // Example, if future blocks is 2, and we are in block 100, it will send a bundle to blocks 102, 103, 104 (assuming a burst size of 3)
+      // and 102 would be the firstBlockOfBatch
+      const firstBlockOfBatch = block.number + FUTURE_BLOCKS;
+
+      // We create our batch of bundles. In this case we use createBundlesWithSameTxs, as all bundles use the same transaction
+      const bundles = createBundlesWithSameTxs({
+        unsignedTxs: txs,
+        burstSize: BURST_SIZE,
+        firstBlockOfBatch,
+      });
+
+      // We send our bundles to Flashbots and retry until the job is worked by us or another keeper.
+      const result = await sendAndRetryUntilNotWorkable({
+        txs,
+        provider,
+        priorityFeeInWei: PRIORITY_FEE,
+        bundles,
+        newBurstSize: BURST_SIZE,
+        flashbots,
+        signer: txSigner,
+        async isWorkableCheck(): Promise<boolean> {
```

Comment:
Here I believe there's an edge case when:
1- Tx sent in bundles is `settleAccount`
2- Another keeper settles it before ours
3- `isWorkableCheck` checks if `canHarvest` returns `true`--it can do so as it's time-based and not based on whether the account needs settlement
4- If `canHarvest` returns `true` the script will keep on trying to send failing bundles with `settleAccount` despite it having been settled until another keeper calls `harvest` and the `isWorkableCheck` starts to fail again.

If this can happen and I'm not missing something, there are two options to fix it:

1- You may want to do the `isWorkableCheck` based on the transaction sent. So add a conditional based on the `isAccountSettlementRequired` variable you have stored. If it's needed this means the tx being sent is `settleAccount`. So, make the `isWorkableCheck` check if the settlement still needs to be settled. If it's not, return `false`, which will restart the script entirely. If it's not needed, it means we are sending `harvest` txs, so it should check for `canHarvest` as it does right now.

2- Divide this script into two. One in charge of working`settleAccount`, the other in charge of working `harvest`. Of course, this is a lot of duplicated code that can be shared, which is not tidy. If going for this approach, we did something similar for `yearn` scripts if you need a reference.

I think option 1) should be more than enough though!